src/SkipStages.cpp

```diff
@@ -18,14 +18,12 @@ using std::map;
 namespace {
 
 bool extern_call_uses_buffer(const Call *op, const std::string &func) {
-   if (op->call_type == Call::Extern) {
-     for (size_t i = 0; i < op->args.size(); i++) {
+    if (op->call_type == Call::Extern) {
+        for (size_t i = 0; i < op->args.size(); i++) {
             const Variable *var = op->args[i].as<Variable>();
-            if (var &&
```

Comment:
It occurs to me that the original code might have been written this way for memoizing tuples. Add a tuple case to your test and see what the buffers are called. You might need to revert the change to this function and instead do something like bail out if it's a Call that starts with halide_memoization_cache_